url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/com/serotonin/bacnet4j/service/confirmed/CreateObjectRequest.java

```diff
@@ -0,0 +1,172 @@
+import com.serotonin.bacnet4j.exception.BACnetException;
+import com.serotonin.bacnet4j.exception.BACnetServiceException;
+import com.serotonin.bacnet4j.obj.BACnetObject;
+import com.serotonin.bacnet4j.service.acknowledgement.AcknowledgementService;
+import com.serotonin.bacnet4j.service.acknowledgement.CreateObjectAck;
+import com.serotonin.bacnet4j.type.Encodable;
+import com.serotonin.bacnet4j.type.ThreadLocalObjectTypeStack;
+import com.serotonin.bacnet4j.type.constructed.Address;
+import com.serotonin.bacnet4j.type.constructed.Choice;
+import com.serotonin.bacnet4j.type.constructed.PropertyValue;
+import com.serotonin.bacnet4j.type.constructed.SequenceOf;
+import com.serotonin.bacnet4j.type.enumerated.ObjectType;
+import com.serotonin.bacnet4j.type.error.CreateObjectError;
+import com.serotonin.bacnet4j.type.primitive.ObjectIdentifier;
+import com.serotonin.bacnet4j.type.primitive.OctetString;
+import com.serotonin.bacnet4j.type.primitive.UnsignedInteger;
+import org.free.bacnet4j.util.ByteQueue;
+
+public class CreateObjectRequest extends ConfirmedRequestService {
+    private static final long serialVersionUID = -610206284148696878L;
+
+    public static final byte TYPE_ID = 10;
+
+    private static List<Class<? extends Encodable>> classes;
+    static {
+        classes = new ArrayList<Class<? extends Encodable>>();
+        classes.add(ObjectType.class);
+        classes.add(ObjectIdentifier.class);
+    }
+
+    private final Choice objectSpecifier;
+    private final SequenceOf<PropertyValue> listOfInitialValues;
+
+    public CreateObjectRequest(ObjectType objectType, SequenceOf<PropertyValue> listOfInitialValues) {
+        objectSpecifier = new Choice(0, objectType);
+        this.listOfInitialValues = listOfInitialValues;
+    }
+
+    public CreateObjectRequest(ObjectIdentifier objectIdentifier, SequenceOf<PropertyValue> listOfInitialValues) {
+        objectSpecifier = new Choice(1, objectIdentifier);
+        this.listOfInitialValues = listOfInitialValues;
+    }
+
+    @Override
+    public byte getChoiceId() {
+        return TYPE_ID;
+    }
+
+    @Override
+    public AcknowledgementService handle(LocalDevice localDevice, Address from, OctetString linkService)
+            throws BACnetErrorException {
+        final ObjectIdentifier id;
+        if (objectSpecifier.getContextId() == 0) {
+            ObjectType type = (ObjectType) objectSpecifier.getDatum();
+            id = localDevice.getNextInstanceObjectIdentifier(type);
+        }
+        else
+            id = (ObjectIdentifier) objectSpecifier.getDatum();
+
+        BACnetObject obj = new BACnetObject(localDevice, id);
+
+        if (listOfInitialValues != null) {
+            for (int i = 0; i < listOfInitialValues.getCount(); i++) {
+                PropertyValue pv = listOfInitialValues.get(i + 1);
+                try {
+                    obj.setProperty(pv);
+                }
+                catch (BACnetServiceException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 109.